Configurations.md

```diff
@@ -1720,6 +1720,14 @@ fn example() {
 }
 ```
 
+## `recursive` 
```

Comment:
We should consider `recursive` as one of the internal options. We do not want users to add this to the configuration by themselves.

Reply:
Will do 👍, though out of curiosity, why should this be a CLI only flag for users?

One downside I see with exposing `recursive` as a config option is that `recursive = false` in the config file would functionally be ignored when using `cargo fmt` since cargo fmt will always pass `--recursive` flag to rustfmt which will always take precedence over the config file value (which could be confusing for users).

Reply:
The options in the configuration file are meant to be persistent, whereas `recursive` is more of a one-shot configuration, which best suits as a CLI flag.

Reply:
Makes sense to me! It wouldn't surprise me if there is a future request to support `recursive` as a config option since `skip_children` was available, but there's clearly several good reasons why `recursive` should be internal/cli-only. 

Side note, while making recursive internal I noticed that internal options added to the config file seem to just be silently ignored. Do I have that correct? 

If they are silently ignored:

Is that the desired behavior or would it beneficial to emit a warning (`"Warning: Unknown configuration option ..`)? Would that potential change be tackled (perhaps it already is) as part of #3873 or should I open a new issue?

Reply:
> Side note, while making recursive internal I noticed that internal options added to the config file seem to just be silently ignored. Do I have that correct?

Yes.

> Is that the desired behavior or would it beneficial to emit a warning ("Warning: Unknown configuration option ..)?

We should make it to a hard error rather than silently ignoring it.

> Would that potential change be tackled (perhaps it already is) as part of #3873 or should I open a new issue?

This should be fixed by #3873.